frame/support/src/dispatch.rs

```diff
@@ -2023,33 +2168,11 @@ macro_rules! decl_module {
 			}
 		}
 
-		// Bring `GetPalletVersion` into scope to make it easily usable.
-		pub use $crate::traits::GetPalletVersion as _;
-		// Implement `GetPalletVersion` for `Module`
-		impl<$trait_instance: $trait_name $(<I>, $instance: $instantiable)?> $crate::traits::GetPalletVersion
-			for $mod_type<$trait_instance $(, $instance)?> where $( $other_where_bounds )*
-		{
-			fn current_version() -> $crate::traits::PalletVersion {
-				$crate::crate_to_pallet_version!()
-			}
-
-			fn storage_version() -> Option<$crate::traits::PalletVersion> {
-				let key = $crate::traits::PalletVersion::storage_key::<
-						<$trait_instance as $system::Config>::PalletInfo, Self
-					>().expect("Every active pallet has a name in the runtime; qed");
-
-				$crate::storage::unhashed::get(&key)
-			}
-		}
-
 		// Implement `OnGenesis` for `Module`
 		impl<$trait_instance: $trait_name $(<I>, $instance: $instantiable)?> $crate::traits::OnGenesis
 			for $mod_type<$trait_instance $(, $instance)?> where $( $other_where_bounds )*
 		{
-			fn on_genesis() {
-				$crate::crate_to_pallet_version!()
-					.put_into_storage::<<$trait_instance as $system::Config>::PalletInfo, Self>();
-			}
+			fn on_genesis() {}
```

Comment:
I think we should do same as in pallet macro: on genesis the storage version is known: it is the storage version of the pallet. So it should be safe to write it.

```suggestion
			fn on_genesis() {
				let storage_version = <Self as $crate::traits::GetStorageVersion>::current_storage_version();
				storage_version.put::<Self>();
			}
```

Reply:
Good find!